hugging-face/runtime/src/main/java/io/quarkiverse/langchain4j/huggingface/QuarkusHuggingFaceChatModel.java

```diff
@@ -0,0 +1,144 @@
+package io.quarkiverse.langchain4j.huggingface;
+
+import static dev.langchain4j.model.huggingface.HuggingFaceModelName.TII_UAE_FALCON_7B_INSTRUCT;
+import static java.util.stream.Collectors.joining;
+
+import java.time.Duration;
+import java.util.List;
+
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.huggingface.client.HuggingFaceClient;
+import dev.langchain4j.model.huggingface.client.Options;
+import dev.langchain4j.model.huggingface.client.Parameters;
+import dev.langchain4j.model.huggingface.client.TextGenerationRequest;
+import dev.langchain4j.model.huggingface.client.TextGenerationResponse;
+import dev.langchain4j.model.huggingface.spi.HuggingFaceClientFactory;
+import dev.langchain4j.model.output.Response;
+
+/**
+ * This is a Quarkus specific version of the HuggingFace model.
+ * <p>
+ * TODO: remove this in the future when the stock {@link dev.langchain4j.model.huggingface.HuggingFaceChatModel}
+ * has been updated to fit our needs (i.e. allowing {@code returnFullText} to be null and making {code accessToken} optional)
+ */
+public class QuarkusHuggingFaceChatModel implements ChatLanguageModel {
+
+    public static final HuggingFaceClientFactory CLIENT_FACTORY = new QuarkusHuggingFaceClientFactory();
+    private final HuggingFaceClient client;
+    private final Double temperature;
+    private final Integer maxNewTokens;
+    private final Boolean returnFullText;
+    private final Boolean waitForModel;
+
+    private QuarkusHuggingFaceChatModel(Builder builder) {
+        this.client = CLIENT_FACTORY.create(new HuggingFaceClientFactory.Input() {
+            @Override
+            public String apiKey() {
+                return builder.accessToken;
+            }
+
+            @Override
+            public String modelId() {
+                return builder.modelId;
+            }
+
+            @Override
+            public Duration timeout() {
+                return builder.timeout;
+            }
+        });
+        this.temperature = builder.temperature;
+        this.maxNewTokens = builder.maxNewTokens;
+        this.returnFullText = builder.returnFullText;
+        this.waitForModel = builder.waitForModel;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public Response<AiMessage> generate(List<ChatMessage> messages) {
+
+        TextGenerationRequest request = TextGenerationRequest.builder()
+                .inputs(messages.stream()
+                        .map(ChatMessage::text)
+                        .collect(joining("\n")))
+                .parameters(Parameters.builder()
+                        .temperature(temperature)
+                        .maxNewTokens(maxNewTokens)
+                        .returnFullText(returnFullText)
```

Comment:
Only if explicitly set.

Reply:
This is `null` by default (in the builder), so setting here does not make a difference.

Reply:
Furthermore, our Quarkus config does not set this either, so it truly is `null` unless the user explicitly configures it